backend/api/models.py

```diff
@@ -2,38 +2,67 @@
 from django.contrib.auth.models import AbstractUser
 
 
-# ID is created automatically
-class Place(models.Model):
-    address = models.CharField(max_length=55, unique=True)
-    name = models.CharField(max_length=55)
+class Account(models.Model):
+    name = models.CharField(max_length=64)
 
 
-class Category(models.Model):
-    name = models.CharField(max_length=55, unique=True)
+class Source(models.Model):
+    account = models.ForeignKey(Account, on_delete=models.CASCADE)
 
 
-class Account(models.Model):
-    name = models.CharField(max_length=55)
-    created = models.DateTimeField(auto_now_add=True)
-    updated = models.DateTimeField(auto_now=True)
+class Private(Source):
+    first_name = models.CharField(max_length=64)
+    last_name = models.CharField(max_length=64)
+
+
+class ZipCity(models.Model):
+    """
+    needed for the Store model to make sure we don't save redundant data
+    (functional dependency from Store.source_id to Zip as well as City, but City is functionally dependent on Zip as well)
+    """
+    zip = models.CharField(max_length=8, primary_key=True)
+    city = models.CharField(max_length=64)
+
+
+class Store(Source):
+    name = models.CharField(max_length=64)
+    street = models.CharField(max_length=64)
+    zip = models.ForeignKey(ZipCity, on_delete=models.CASCADE)
+    house_number = models.CharField(max_length=64)
+
+
+class Category(models.Model):
+    name = models.CharField(max_length=64, unique=True)
+    account = models.ForeignKey(Account, on_delete=models.CASCADE)
 
 
 class Cashflow(models.Model):
     is_income = models.BooleanField()
     overall_value = models.DecimalField(max_digits=6, decimal_places=2)
     created = models.DateTimeField(auto_now_add=True)
     updated = models.DateTimeField(auto_now=True)
-    category = models.ForeignKey(Category, on_delete=models.DO_NOTHING)
-    place = models.ForeignKey(Place, on_delete=models.DO_NOTHING, blank=True, null=True)
+    category = models.ForeignKey(Category, on_delete=models.CASCADE)
+    source = models.ForeignKey(
+        Source, on_delete=models.SET_NULL, blank=True, null=True)
```

Comment:
This is not how it works, is it? Here I declare that whenever a Cashflow is deleted, the Source should be kept and not deleted as well. So, in the future, you can easily reselect the place, because you had a cashflow there in the past.

Reply:
I think it means, whenever a Source will be deleted, the cashflow will still be existent. It will then have a source = null value stored. As visible in the EER, it is not allowed to have an optional store.

Reply:
Thanks for the heads up, I had an error in thought. It is fixed here: https://github.com/Cebox82/FiBo/commit/ec40c89f9e236c78839f83252f86acecd45fad11

Reply:
(At least I hope so! Please verify I fixed the issue as you'd expect.)